packages/babel-plugin-transform-spread/src/index.js

```diff
@@ -50,11 +50,11 @@ export default function({ types: t }, options) {
         if (!hasSpread(elements)) return;
 
         const nodes = build(elements, scope, state);
-        let first = nodes.shift();
+        const first = nodes.shift();
```

Comment:
could u explain the changed part and logic behind it? i just havent read properly through the plugin's code and I'd appreciate a quick explanation to catch up

I'd also expect more tests to be affected by your desired change 🤔 seems strange that there are only 5

Reply:
So, basically before this, if `first` was not an `ArrayExpression` then a new array was created and that was the starting point for calling `concat`. But because of how `toArray` works, this is actually not necessary.

The first simplification here is just getting rid of the extra empty array that starts the concat chain. That's pretty straightforward.

The second simplification here is not even calling `concat` if there's only 1 element and that element is not equivalent to the one we started with. This relies on reading the logic in `toArray` which for arrays is just an identity function (returns the node passed to it) and otherwise uses a helper to create a new array. So if `elements[0].argument !== first` then we know that either `Array.prototype.slice.call` or `toConsumableArray` were applied.

This only affects a few different situations (hence how few tests are affected):

1) If there's only 1 element which is not an ArrayExpression
2) If there are multiple elements but the first element is not an ArrayExpression.

Reply:
Cool, sounds reasonable